src/main/java/io/takari/aether/wagon/OkHttpWagon.java

```diff
@@ -78,7 +80,7 @@ public void fillInputData(InputData inputData) throws TransferFailedException, R
         throw new TransferFailedException("Failed to transfer file: " + url + ". Return code is: " + statusCode);
       }
       
-      inputData.setInputStream(response.getInputStream());
+      inputData.setInputStream(new ByteArrayInputStream(response.getBytes()));
```

Comment:
why is this necessary?

Reply:
Otherwise WagonTest.testNoHeaders will fail.

Reply:
This is indeed incorrect, `fillInputData` cannot close http response, it must return live input stream to the caller (via `InputData` parameter field) and rely on the caller to close the stream.